cmd/keymasterd/config.go

```diff
@@ -327,33 +333,33 @@ func loadVerifyConfigFile(configFilename string) (*RuntimeState, error) {
 	if strings.HasPrefix(string(runtimeState.SSHCARawFileContent[:]), "-----BEGIN RSA PRIVATE KEY-----") {
 		signer, err := getSignerFromPEMBytes(runtimeState.SSHCARawFileContent)
 		if err != nil {
-			logger.Printf("Cannot parse Priave Key file")
+			logger.Printf("Cannot parse Private Key file")
 			return nil, err
 		}
 		runtimeState.caCertDer, err = generateCADer(&runtimeState, signer)
 		if err != nil {
-			logger.Printf("Cannot generate CA Der")
+			logger.Printf("Cannot generate CA DER")
 			return nil, err
 		}
-
-		// Assignmet of signer MUST be the last operation after
+		// Assignment of signer MUST be the last operation after
 		// all error checks
 		runtimeState.Signer = signer
 		runtimeState.signerPublicKeyToKeymasterKeys()
 		runtimeState.SignerIsReady <- true
 
 	} else {
-		if runtimeState.ClientCAPool == nil {
-			err := errors.New("Invalid ssh CA private key file and NO clientCA")
-			return nil, err
-		}
-		//check that the loaded date seems like an openpgp armored file
+		//check that the loaded data seems like an openpgp armored file
 		fileAsString := string(runtimeState.SSHCARawFileContent[:])
 		if !strings.HasPrefix(fileAsString, "-----BEGIN PGP MESSAGE-----") {
-			err = errors.New("Have a client CA but the CA file does NOT look like and PGP file")
-			return nil, err
+			return nil, errors.New("CA private key file does NOT look like PGP")
+		}
+		logger.Println("Starting up in sealed state")
+		if runtimeState.ClientCAPool == nil {
+			logger.Println("No client CA: manual unsealing not possible")
+		}
+		if err := runtimeState.tryAutoUnseal(); err != nil {
```

Comment:
Why is this blocking? I think this should be goroutine, that IF detects is on AWS will keep retrying until the system is unsealed.

Reply:
I did it this way because it's really fast. However, I've changed it to a loop so that if there is a policy error or some such, Keymaster will keep trying until success.